### PR TITLE
docker: use /host/etc/hostname if mounted

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

Fixes: #16396

##### Test Plan

Start docker container with/without mounted /host/etc/hostname and check the hostname in `/netdata.conf`.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, LGTM!

File: daemon/main.c [hunk 1]
```diff
@@ -1179,9 +1182,6 @@ static void get_netdata_configured_variables() {
 #endif
     // ------------------------------------------------------------------------
```
Comment (bdkuhman) [inline daemon/main.c:1182]: lint from function movement

**Review by bdkuhman — APPROVED** [review]
Working as expected! Thank you so much!

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, LGTM!

**Review by MrZammler — APPROVED** [review]

**Pull request merged**